### DISCUSSION TITLE
Finding the line heatmap pixels that corresponds to a line segment : SOLD2 model tutorial (Line Detection and Matching)

### DISCUSSION THREAD

**Discussion by sunshineinsandiego** [kornia/kornia#2135] [Questions]
Hi - I am using the SOLD2 model for line detection and matching based on the tutorial [here.](https://kornia-tutorials.readthedocs.io/en/latest/line_detection_and_matching_sold2.html) How can I retrieve the parts of the line heatmap (i.e. the `(x,y)` coordinates of the pixels from `outputs['line_heatmap']`) that correspond with the final reported line segments (`outputs['line_segments']`)?

For example: If `outputs['line_segments']` returns a line segment with endpoints `[(y1, x1), (y2, x2)]`, how can I find the heatmap pixels that correspond with that line segment? 

My understanding of the returned information from `outputs` is:
- The `line_heatmap` may include detected curves/edges that are not necessarily completely straight line segments.
- The `line_heatmap` values are used in conjunction with the `junction_heatmap` values to determine potential line segments.
- The final `line_segments` that are returned by the model are straight line segments parameterized by two endpoints.

So given two reported endpoints of a line segment from `outputs['line_segments']`, can I retrieve the associated pixels from `outputs['line_heatmap']` that correspond to that line segment?

Is it always true that the two reported endpoints of a line segment from `outputs['line_segments']` will also be on a line in the `line_heatmap` (i.e. the coordinates of the endpoints will have a high pixel intensity in the heatmap)?

Thanks!

**Comment by edgarriba** [comment]
I’d suggest to use `kornia.geometry.remap` or similar to extract values from coordinates. cc/ @ducha-aiki 

* https://kornia.readthedocs.io/en/latest/geometry.transform.html#kornia.geometry.transform.remap

**Reply by sunshineinsandiego** [reply]
My question is more about the architecture / outputs of SOLD2. Is it possible to output each of the heatmap lines that are used to define each detected and matched line segment?
